lib/statsd/instrument/assertions.rb

```diff
@@ -31,41 +31,56 @@ def assert_statsd_key_value(metric_name, options = {}, &block)
     assert_statsd_call(:kv, metric_name, options, &block)
   end
 
-  private
-
-  def assert_statsd_call(metric_type, metric_name, options = {}, &block)
-    options[:times] ||= 1
+  def assert_statsd_calls(expected_metrics, &block)
     metrics = capture_statsd_calls(&block)
-    metrics = metrics.select { |m| m.type == metric_type && m.name == metric_name }
-    assert metrics.length > 0, "No StatsD calls for metric #{metric_name} were made."
-    assert options[:times] === metrics.length, "The amount of StatsD calls for metric #{metric_name} was unexpected. Expected #{options[:times].inspect}, found #{metrics.length}"
 
-    metrics.each do |metric|
+    expected_metrics.each do |expected_metric|
+      matching_metrics = metrics.select { |m| m.type == expected_metric.type && m.name == expected_metric.name }
+      assert matching_metrics.length > 0, "No StatsD calls for metric #{expected_metric.name} were made."
 
-      assert within_numeric_range?(metric.sample_rate), "Unexpected sample rate type for metric #{metric_name}, must be numeric"
-      assert_equal options[:sample_rate], metric.sample_rate, "Unexpected StatsD sample rate for metric #{metric_name}" if options[:sample_rate]
-      assert_equal options[:value], metric.value, "Unexpected value submitted for StatsD metric #{metric_name}" if options[:value]
+      matching_metrics.each do |actual_metric|
+        assert_metric_matches_expectation(actual_metric, expected_metric)
+      end
 
-      if options[:tags]
+      assert expected_metric.times == matching_metrics.length,
+        "The amount of StatsD calls for metric #{expected_metric.name} was unexpected. Expected #{expected_metric.times}, found #{matching_metrics.length}"
+    end
+  end
 
-        expected_tags = Set.new(StatsD::Instrument::Metric.normalize_tags(options[:tags]))
-        actual_tags = Set.new(metric.tags)
+  private
 
-        if options[:ignore_tags]
-          ignored_tags = Set.new(StatsD::Instrument::Metric.normalize_tags(options[:ignore_tags])) - expected_tags
-          actual_tags -= ignored_tags
+  def assert_metric_matches_expectation(actual_metric, expected_metric)
```

Comment:
I feel like this method might just as well be an instance method on the expectation class, so you can just say `expected_metric.matches?(actual_metric)` and that returns true or false (and maybe sets some error message or something).